### PR TITLE
feat: refactor ts module internal to be more clear

### PR DESCRIPTION
**Pull request by TomChv**
Improve the TypeScript module internal to be easier to maintain and more clear.
This refactor aims to abstract the scan logic using classes to hide the complexity and expose a simpler interface to interact with. By doing so, we no longer need standalone functions to do the introspection, the class is responsible for his own introspection.

After completing the refactor, most of the scan old code with be replaced by the new logic.

This refactor also aims to simplify some logic with the TypeScript compiler API.

- [x] Param
- [x] Method (function)
- [x] Constructor
- [x] Property
- [x] Class
- [x] Module
- [x] Module description

In some follow up PRs I also plan to:
- Update the `register` functions by implementing `register` at the abstraction level which I think will make things way simpler. -> In a follow up PR, this one is already big enough
- Refactor some invoke logic to use the abstraction (which will make the code waaaaaaaay simpler specially when I do check for the arg order and variadic) -> Done ✅ 
- Improve the testing scan process to be tested against manifest instead of manual JSON structure. This will improve the TS SDK scalability when we'll support new features) -> Done ✅ 
  By doing so, we'll also drastically improve simplicity of the tests AND also allow us to test some specific part of the introspection without any complexity.

There's still a long way to go, this PR is also a first step in my long journey :rocket:

⚠️ Even if it's a big refactor! It's not going to impact users, it's only internal 

Fixes DEV-3515

### CODE REVIEW

**Review by gerhard — COMMENTED** [review]
This looks good to me! WDYT @helderco?

**Review by helderco — APPROVED** [review]

File: sdk/typescript/introspector/scanner/abtractions/method.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    const expression = this.decorator.expression as ts.CallExpression
+    const aliasArg = expression.arguments[0]
+
+    if (!aliasArg) {
+      return undefined
+    }
+
+    return JSON.parse(aliasArg.getText().replace(/'/g, '"'))
```
[the remark this answers is not in the document]
Comment (TomChv) [reply sdk/typescript/introspector/scanner/abtractions/method.ts:98]: `@func('foo')`

File: sdk/typescript/introspector/scanner/abtractions/argument.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      isOptional: this.isOptional,
+      defaultValue: this.defaultValue,
+    }
+  }
+
+  /**
+   * The TypeScript Compiler API returns the raw default value as it is written
+   * by the user.
+   * However, some notations are not supported by GraphQL so this function
+   * formats the default value to be compatible with the GraphQL syntax.
```
[the remark this answers is not in the document]
Comment (TomChv) [reply sdk/typescript/introspector/scanner/abtractions/argument.ts:167]: Good catch, I'll take care of that in a follow up

File: sdk/typescript/introspector/scanner/abtractions/method.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (helderco) [reply sdk/typescript/introspector/scanner/abtractions/method.ts:98]: Oh really? 😄

**Pull request merged**